syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -323,15 +309,15 @@ public int compareTo(Path other) {
 
     @Override
     public boolean equals(Object ob) {
-        if ((ob != null) && (ob instanceof Path)) {
+        if (ob instanceof Path) {
             return compareTo((Path) ob) == 0;
         }
         return false;
     }
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(segments.toArray());
```

Comment:
removed an unnecessary array copy allocation, add unit test confirming hashCode backward compatibility (though shouldn't really be necessary)